File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDOFuncSimpleFileOptParamModel.swift
//
//  IDOFuncSimpleFileOptParamModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/12.
//

import Foundation
/// Simple file operations event
@objcMembers
public class IDOFuncSimpleFileOptParamModel: NSObject, IDOBaseModel {
    /// Operation type
    /// 0: Get
    /// 1: Overwrite
    /// 2: Delete
    /// 3: Copy
    public var operate: Int
    /// Index number
    public var index: Int
    /// Destination index, only used for copying, invalid for other cases
    public var destIndex: Int

    enum CodingKeys: String, CodingKey {
        case operate = "operate"
        case index = "index"
        case destIndex = "dest_index"
    }

    public init(operate: Int, index: Int, destIndex: Int) {
        self.operate = operate
        self.index = index
        self.destIndex = destIndex
    }

    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDOFuncSimpleFileOptParamModel.kt
//
//  IDOfuncSimpleFileOptParamModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

///
/// Simple file operations event

open class IDOFuncSimpleFileOptParamModel(operate: Int, index: Int, destIndex: Int) : IDOBaseModel {

    /// Operation type
    /// 0: Get
    /// 1: Overwrite
    /// 2: Delete
    /// 3: Copy
    @SerializedName("operate")
    var operate: Int = operate

    /// Index number
    @SerializedName("index")
    var index: Int = index

    /// Destination index, only used for copying, invalid for other cases
    @SerializedName("dest_index")
    var destIndex: Int = destIndex


    

    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}
    